mainMARS.s

```diff
@@ -217,12 +217,57 @@ max:
   pop_ra_and_return
 
 # Power operation.
+.data 
+power_a_message: .asciiz "Enter the a in (a ^ b):\n"
+power_b_message: .asciiz "Enter the b in (a ^ b):\n"
+zero_exponent_message: .asciiz "The result equals 1 " 		#Special_Case if exponent equals zero,Then the result equals one
 
 .text
-power:
-  push_ra
-  la REG_PRINT_STRING_ARG, unimplemented_message
+power:  
+  # Print the message for a.
+  la REG_PRINT_STRING_ARG, power_a_message
+  jal printString
+
+  # Read a.
+  jal readInteger
+  move $t1, REG_READ_INTEGER_RET
+
+  # Print the message for b.
+  la REG_PRINT_STRING_ARG, power_b_message
+  jal printString
+
+  # Read b.
+  jal readInteger
+  move $t2, REG_READ_INTEGER_RET
```

Comment:
Same as above. Also, this value maybe negative, which is a case you don't consider.